### PR TITLE
Upgrade MySQL to v8.0

### LINKED ISSUE #259: Migrate from MySQL 5.6 -> 8.0
Try to upgrade from MySQL to MySQL 8.0 and run the *full migrations* + tests.

Expected achievements:
* *IF* the upgrade requires some data or schema migration before, create the migration script or query under `db/`.
* *IF* it requires some data or schema migration after (because some tests do not pass), make sure to add them (may be required to put it as first migration)
* Make sur the docker-compose and circle-ci config are correctly updated

### PR DESCRIPTION
**Pull request by zenovich**
Fixes #259

### CODE REVIEW

File: app/database/user_item_store_compute_all_user_items.go [hunk 1]
```diff
@@ -120,15 +120,20 @@ func (s *UserItemStore) ComputeAllUserItems() (err error) {
 					LEFT JOIN items_items
 						ON items_items.idItemParent = users_items.idItem
 					SET
-						users_items.sLastActivityDate = IF(task_children_data.idUserItem IS NOT NULL AND items_items.ID IS NOT NULL,
+						users_items.sLastActivityDate = IF(task_children_data.idUserItem IS NOT NULL AND
```
Comment (smadbe) [inline app/database/user_item_store_compute_all_user_items.go]: I don't get the need for this change. Can you tell me more about it?

File: app/service/utils.go [hunk 2]
```diff
@@ -264,9 +264,21 @@ func setConvertedValueToJSONMap(valueName string, value interface{}, result map[
 		value = strconv.FormatInt(valueInt64, 10)
 	}
 
+	value = convertDateToRFC3339(value, snakeCaseName)
 	result[snakeCaseName] = value
 }
 
+func convertDateToRFC3339(value interface{}, snakeCaseName string) interface{} {
```
Comment (smadbe) [inline app/service/utils.go]: Wouldn't `convertToRFC3339IfDate` be more appropriate as name? From what I understand, you do not only give dates as args.

File: app/database/user_item_store_compute_all_user_items.go (hunk 1 above)
Comment (zenovich) [reply app/database/user_item_store_compute_all_user_items.go]: We modify fields depending on `children_data` only if `children_data` row exists. We didn't check that explicitly earlier and it worked somewhat correctly in MySQL 5.6, but not in MySQL 8.0

File: app/api/items/get_item.go [hunk 3]
```diff
@@ -60,21 +60,21 @@ type itemUser struct {
...
+	Type                   string         `json:"type"`
+	DisplayDetailsInParent bool           `json:"display_details_in_parent"`
+	ValidationType         string         `json:"validation_type"`
+	HasUnlockedItems       bool           `json:"has_unlocked_items"` // whether items.idItemUnlocked is empty
+	ScoreMinUnlock         int32          `json:"score_min_unlock"`
+	TeamMode               *string        `json:"team_mode"`
+	TeamsEditable          bool           `json:"teams_editable"`
+	TeamMaxMembers         int32          `json:"team_max_members"`
+	HasAttempts            bool           `json:"has_attempts"`
+	AccessOpenDate         *database.Time `json:"access_open_date"` // iso8601 str
```
[the remark this answers is not in the document]
Comment (zenovich) [reply app/api/items/get_item.go]: fixed

File: app/api/currentuser/refresh.go [hunk 4]
```diff
@@ -30,6 +31,9 @@ func (srv *Service) refresh(w http.ResponseWriter, r *http.Request) service.APIE
 	service.MustNotBeError(err)
 
 	userProfile["sLastActivityDate"] = database.Now()
+	if defaultLanguage, ok := userProfile["sDefaultLanguage"]; ok && defaultLanguage == nil {
+		userProfile["sDefaultLanguage"] = gorm.Expr("DEFAULT")
```
[the remark this answers is not in the document]
Comment (zenovich) [reply app/api/currentuser/refresh.go]: fixed

File: app/database/time.go [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+}
+
+// ScanString assigns a value from string with a database driver value
+func (t *Time) ScanString(str string) (err error) {
+	const timeFormat = "2006-01-02 15:04:05.999999"
+
+	// Based on go-sql-driver/mysql.parseDateTime (see https://github.com/go-sql-driver/mysql/blob/master/utils.go#L109)
+	base := "0000-00-00 00:00:00.0000000"
+	switch len(str) {
+	case 10, 19, 21, 22, 23, 24, 25, 26: // up to "YYYY-MM-DD HH:MM:SS.MMMMMM"
```
[the remark this answers is not in the document]
Comment (zenovich) [reply app/database/time.go]: Is it possible to get time without seconds from MySQL? Looks like it's not. `go-sql-driver/mysql` doesn't parse the format of length 16.

File: app/database/time_test.go [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+)
+
+func TestTime_ScanString(t *testing.T) {
+	tests := []struct {
+		name     string
+		str      string
+		wantErr  bool
+		wantTime time.Time
+	}{
+		{name: "normal datetime", str: "2019-05-30 11:30:12",
```
[the remark this answers is not in the document]
Comment (zenovich) [reply app/database/time_test.go]: fixed

File: app/service/utils.go (hunk 2 above)
Comment (zenovich) [reply app/service/utils.go]: fixed

**Pull request merged**